### ISSUE TITLE
Pluggable `get_document`

### ISSUE THREAD

**Issue by trollfot** [jacksmith15/json-ref-dict#22]
It would be interesting to have a pluggable `get_document` function, to allow fetching schemas from other stores, like local registries. The tests have a similar pattern where documents are resolved from a dict.

**Comment by jacksmith15** [comment]
Could you provide an example of what you mean by local registries?. If I understand correctly - you'd like to be able to override how get_document retrieves the document?

I can certainly see some value in that! Schema's which require authentication to reach are a good example of something not currently supported.

One API option off the top of my head:

```python
from json_ref_dict import loader

@loader.register
def get_document(base_uri: str):
    if meets_condition(base_uri):
        return get_schema_with_special_headers(base_uri)
    return None # Or raise some specific exception - something which causes fallback to default
```

Other options might be to hook this into something which can be subclassed, or overriden within a context manager, as that would make it easier to have different overrides within the same process.

I'd happily accept pull requests to add this feature.

**Comment by jacksmith15** [comment]
I've released [0.7.0](https://pypi.org/project/json-ref-dict/0.7.0/) which contains the PR you submitted - thanks again!
